### DISCUSSION TITLE
Prefixing components

### DISCUSSION THREAD

**Discussion by jonerickson** [themesberg/flowbite-vue#174] [Ideas]
How do you add TW prefixes to the components? I know you can define the TW prefix in the TW config however this does not force the components to be built with the same prefixes rendering it useless. How might one add the TW prefixes? Thanks.

**Comment by acidiney** [comment]
When you import components, you can rename the component with prefixes.

```vue
<script setup>
import { Textarea as TwTextarea } from 'flowbite-vue'
</script>

<template>
  <tw-textarea label="something" />
</template>
```
It will need some extra work, but I think it is the only way to prefix.

**Comment by jonerickson** [comment]
I am walking about the TW CSS classes when it builds, not the flow bite modules. Thanks. In TW you can add a prefix to the compiled classes in the TW config.

**Comment by cogor** [comment]
Hey there! Currently, our components can't handle prefixes. However, I'm actively researching ways to implement this feature. Though I don't have a specific timeline yet, rest assured that I'm working on it.

**Comment by jonerickson** [comment]
> Hey there! Currently, our components can't handle prefixes. However, I'm actively researching ways to implement this feature. Though I don't have a specific timeline yet, rest assured that I'm working on it.

Great, thanks for response.

**Comment by jonerickson** [comment]
The reason I ask is, I am building some react components that are embeddable widgets that can be embedded in another TW website and I don’t want the styles to clash.

**Comment by Sqrcz** [comment]
This closely relates to https://github.com/themesberg/flowbite-vue/issues/434

Currently not planned for the near future.
